packages/graphql/test/enumtypedefinitions.test.tsx

```diff
@@ -0,0 +1,347 @@
+        ACTIVE
+        INACTIVE
+      }
+    `);
+  });
+
+  it("renders an enum with multiple directives", () => {
+    const result = toGraphQLText(
+      <gql.EnumTypeDefinition
+        name="Status"
+        directives={
+          <>
+            <gql.Directive name="auth" args={{ requires: "ADMIN" }} />
+            <gql.Directive name={builtInDirectives.deprecated} />
+          </>
+        }
+      >
+        <gql.EnumValue name="ACTIVE" />
+      </gql.EnumTypeDefinition>,
+    );
+    expect(result).toRenderTo(d`
+      enum Status @auth(requires: "ADMIN") @deprecated {
+        ACTIVE
+      }
+    `);
+  });
+
+  it("renders an enum value with a directive", () => {
+    const result = toGraphQLText(
+      <gql.EnumTypeDefinition name="Status">
+        <gql.EnumValue name="ACTIVE" />
+        <gql.EnumValue
+          name="DEPRECATED_STATUS"
+          directives={
+            <gql.Directive
+              name={builtInDirectives.deprecated}
+              args={{ reason: "Use INACTIVE instead" }}
+            />
+          }
+        />
+        <gql.EnumValue name="INACTIVE" />
+      </gql.EnumTypeDefinition>,
+    );
+    expect(result).toRenderTo(d`
+      enum Status {
+        ACTIVE
+        DEPRECATED_STATUS @deprecated(reason: "Use INACTIVE instead")
+        INACTIVE
+      }
+    `);
+  });
+
+  it("supports enum references in field types", () => {
+    const statusRef = refkey();
+
+    const result = toGraphQLText(
+      <>
+        <gql.EnumTypeDefinition name="Status" refkey={statusRef}>
+          <gql.EnumValue name="ACTIVE" />
+          <gql.EnumValue name="INACTIVE" />
+        </gql.EnumTypeDefinition>
+        <gql.ObjectTypeDefinition name="User">
+          <gql.FieldDefinition name="status" type={code`${statusRef}!`} />
+        </gql.ObjectTypeDefinition>
+      </>,
+    );
+
+    expect(result).toRenderTo(d`
+      enum Status {
+        ACTIVE
+        INACTIVE
+      }
+      
+      type User {
+        status: Status!
+      }
+    `);
+  });
+
+  it("supports cross-file enum references", () => {
+    const statusRef = refkey();
+
+    const res = toGraphQLTextMultiple([
+      <gql.SourceFile path="enums.graphql">
+        <gql.EnumTypeDefinition name="Status" refkey={statusRef}>
+          <gql.EnumValue name="ACTIVE" />
+          <gql.EnumValue name="INACTIVE" />
+          <gql.EnumValue name="PENDING" />
+        </gql.EnumTypeDefinition>
+      </gql.SourceFile>,
+      <gql.SourceFile path="types.graphql">
+        <gql.ObjectTypeDefinition name="User">
+          <gql.FieldDefinition name="status" type={code`${statusRef}!`} />
+        </gql.ObjectTypeDefinition>
+      </gql.SourceFile>,
+    ]);
+
+    assertFileContents(res, {
+      "enums.graphql": `
+        enum Status {
+          ACTIVE
+          INACTIVE
+          PENDING
+        }
+      `,
+      "types.graphql": `
+        type User {
+          status: Status!
+        }
+      `,
+    });
+  });
```

Comment:
As you noted [elsewhere](https://github.com/pinterest/alloy/blob/ebc8eb40696bca6b8b09ce7ac57a3930414f7004/packages/graphql/src/symbols/graphql-module-scope.ts#L7-L13), GraphQL does not have native import / cross-module references.